product-service/serverless.ts

```diff
@@ -0,0 +1,184 @@
+import { newProductSchema } from "./src/handlers/createProduct/handler";
+
+const serverlessConfigurationProductService = {
+  service: 'product-service',
+  provider: {
+    name: 'aws',
+    runtime: 'nodejs20.x',
+    stage: 'dev',
+    region: 'eu-west-1',
+    environment: {
+      PRODUCTS_TABLE: 'Products',
+      STOCK_TABLE: 'Stock',
+      REGION: 'eu-west-1',
+      SNS_ARN: 'arn:aws:sns:eu-west-1:637423586618:createProductTopic'
+    },
+    iamRoleStatements: [
+      {
+        Effect: 'Allow',
+        Action: ['sqs:ReceiveMessage', 'sqs:DeleteMessage', 'sqs:GetQueueAttributes'],
+        Resource: {
+          "Fn::GetAtt": [
+            "catalogItemsQueue",
+            "Arn"
+          ]
+        },
+      },
+      {
+        Effect: 'Allow',
+        Action: ['sns:Publish'],
+        Resource: {
+          "Fn::Join": [":", ["arn:aws:sns", { "Ref": "AWS::Region" }, { "Ref": "AWS::AccountId" }, "createProductTopic"]]
+         },
+      },
+      {
+        Effect: 'Allow',
+        Action: [
+          'dynamodb:Scan',
+          'dynamodb:GetItem',
+          'dynamodb:PutItem',
+          'dynamodb:UpdateItem',
+          'dynamodb:DeleteItem',
+          's3:PutObject',
+          's3:GetObject',
+          's3:ListObject',
+          's3:DeleteObject',
+          's3:PutObjectAcl',
+        ],
+        Resource: [
+          'arn:aws:dynamodb:${self:provider.region}:*:table/Products',
+          'arn:aws:dynamodb:${self:provider.region}:*:table/Stock',
+        ],
+      }
+    ]
+  },
+  plugins: [
+    'serverless-offline',
+    'serverless-esbuild'
+  ],
+  custom: {
+    esbuild: {
+      bundle: true,
+      minify: true,
+      sourcemap: true,
+      keepNames: true,
+      tsconfig: 'tsconfig.json',
+      platform: 'node',
+    },
+  },
+  resources: {
+    Resources: {
+      catalogItemsQueue: {
+        Type: 'AWS::SQS::Queue',
+        Properties: {
+          QueueName: 'catalogItemsQueue'
+        }
+      }, 
+      NewProductTopic: {
+        Type: 'AWS::SNS::Topic',
+        Properties: {
+          TopicName: 'createProductTopic'
+        }
+      },
+      NewProductSubscription: {
```

Comment:
Lets add this as well:

+15 (All languages) - set a Filter Policy for SNS createProductTopic in serverless.yml and create an additional email subscription to distribute messages to different emails depending on the filter for any product attribute

Reply:
done